li-hadoop-plugin/src/main/groovy/com/linkedin/gradle/liazkaban/LiAzkabanDslCompiler.groovy

```diff
@@ -62,26 +63,61 @@ class LiAzkabanDslCompiler extends AzkabanDslCompiler {
       return;
     }
 
-    // since it is a hadoopShell job type, only two properties are relevant.
-    Set<String> relevantProperties = ['type','command'];
+    Map<String,String> azkabanOptions = new HashMap<String,String>();
+    azkabanOptions.put(LiAzkabanJavaProperties.AZKABAN_LINK_WORKFLOW_URL,"\${${LiAzkabanJavaProperties.AZKABAN_LINK_WORKFLOW_URL}}")
+    azkabanOptions.put(LiAzkabanJavaProperties.AZKABAN_LINK_ATTEMPT_URL,"\${${LiAzkabanJavaProperties.AZKABAN_LINK_ATTEMPT_URL}}")
+    azkabanOptions.put(LiAzkabanJavaProperties.AZKABAN_LINK_JOB_URL,"\${${LiAzkabanJavaProperties.AZKABAN_LINK_JOB_URL}}")
+    azkabanOptions.put(LiAzkabanJavaProperties.AZKABAN_LINK_EXECUTION_URL,"\${${LiAzkabanJavaProperties.AZKABAN_LINK_EXECUTION_URL}}")
+    azkabanOptions.put(LiAzkabanJavaProperties.AZKABAN_JOB_INNODES,"\${${LiAzkabanJavaProperties.AZKABAN_JOB_INNODES}}")
+    azkabanOptions.put(LiAzkabanJavaProperties.AZKABAN_JOB_OUTNODES,"\${${LiAzkabanJavaProperties.AZKABAN_JOB_OUTNODES}}")
+
+    // create jvm string of the form -Dkey1=value1 -Dkey2=value2
+    StringBuffer azkabanOpts = new StringBuffer();
+    azkabanOptions.each {
+      key,value -> azkabanOpts.append("-D${key}=${value}");
+        azkabanOpts.append(" ");
+    }
+    String azkabanJvmString =  azkabanOpts.toString();
+
+    // add the azkabanjvmString to PIG_JAVA_OPTS to add them to job conf.
+    if(allProperties.hasProperty(PIG_JAVA_OPTS)) {
+      allProperties.put(PIG_JAVA_OPTS, allProperties.get(PIG_JAVA_OPTS) + " " + azkabanJvmString);
+    } else {
+      allProperties.put(PIG_JAVA_OPTS, azkabanJvmString);
+    }
+    // since it is a hadoopShell job type, only some of the properties are relevant.
     String fileName = job.buildFileName(this.parentScope);
 
     File file = new File(this.parentDirectory, "${fileName}.job");
     List<String> sortedKeys = sortPropertiesToBuild(allProperties.keySet());
+    List<String> filteredKeys = new ArrayList<String>();
+    Set<String> irrelevantProperties = ['pig.script','pig.home',/param.*/,'use.user.pig.jar',/hadoop-inject.*/];
+
+    for(String key: sortedKeys) {
+      boolean isRelevant = true;
+      for(String property: irrelevantProperties) {
```

Comment:
Anant - irrelevantProperties is a set! Instead of iterating the set, just use the contains method to determine if the set contains the key